@shared/api/external.ts

```diff
@@ -105,59 +112,79 @@ export const submitBlob = async (
       type: EXTERNAL_SERVICE_TYPES.SUBMIT_BLOB,
     });
   } catch (e) {
-    console.error(e);
-    throw e;
+    return {
+      signedMessage: "",
+      signerAddress: "",
+      error: FreighterApiInternalError,
+    };
   }
-  const { signedBlob, error } = response;
+  const { signedBlob, signerAddress } = response;
 
-  if (error) {
-    throw error;
-  }
-  return signedBlob;
+  return {
+    signedMessage: signedBlob,
+    signerAddress,
+    error: response?.apiError,
+  };
 };
 
 export const submitAuthEntry = async (
   entryXdr: string,
   opts?: {
-    accountToSign?: string;
+    address?: string;
+    networkPassphrase?: string;
   },
-): Promise<string> => {
-  let response = { signedAuthEntry: "", error: "" };
+): Promise<{
+  signedAuthEntry: string;
+  signerAddress: string;
+  error?: FreighterApiError;
+}> => {
   const _opts = opts || {};
-  const accountToSign = _opts.accountToSign || "";
+  const accountToSign = _opts.address || "";
+  let response;
   try {
     response = await sendMessageToContentScript({
       entryXdr,
       accountToSign,
+      networkPassphrase: opts?.networkPassphrase,
       type: EXTERNAL_SERVICE_TYPES.SUBMIT_AUTH_ENTRY,
     });
   } catch (e) {
     console.error(e);
+    return {
+      signedAuthEntry: "",
+      signerAddress: "",
+      error: FreighterApiInternalError,
+    };
   }
-  const { signedAuthEntry, error } = response;
+  const { signedAuthEntry, signerAddress } = response;
 
-  if (error) {
-    throw error;
-  }
-  return signedAuthEntry;
+  return { signedAuthEntry, signerAddress, error: response?.apiError };
 };
 
-export const requestNetwork = async (): Promise<string> => {
-  let response = { network: "", error: "" };
+export const requestNetwork = async (): Promise<{
+  network: string;
+  networkPassphrase: string;
+  error?: FreighterApiError;
+}> => {
+  let response;
   try {
     response = await sendMessageToContentScript({
-      type: EXTERNAL_SERVICE_TYPES.REQUEST_NETWORK,
+      type: EXTERNAL_SERVICE_TYPES.REQUEST_NETWORK_DETAILS,
```

Comment:
this method now return the network and the network passphrase. I'm opting to use the `REQUEST_NETWORK_DETAILS` service so i can grab both those values at once.

`REQUEST_NETWORK`, though not used, will need to stay in the codebase for older versions of freighter-api that still make use of it